### ISSUE TITLE
Koreader crashes on startup | Motorola Moto G6 (ali) XT1925-5 Android 9 SDK28

### ISSUE THREAD

**Issue by saasasasasa** [koreader/koreader#8181]
* KOReader version: koreader-android-arm-v2021.08
* Device: Motorola Moto G6 (ali) XT1925-5 (running on android 9)

#### Issue
Koreader crashes on startup.
#### Steps to reproduce
Launch app on this device.
##### Crash log
[crash.txt](https://github.com/koreader/koreader/files/7110337/crash.txt)

It crashes on startup since version 05. The last working version for me is 2021.04 "Gouty Gauss". Also I had some problems with getting the crash log. After uninstalling version 04 (because android couldn't update it) and installing the newest one, it crashed on startup but the log was completely empty. Workaround was to uninstall, and get the version 06, that's when log was generating fine. Then I updated it to the version 08 and crash log finally shows up.

**Comment by NiLuJe** [comment]
I'm assuming this possibly has to do with the RAM check and Android, but I didn't have time to look into it further (e.g., not sure if there's a nil, or if the computation is so borked that we end up with a floor'ed 0 for some reason).

**Comment by NiLuJe** [comment]
I'd be mildly intrigued about what a debug log might be saying about the cache size, for instance ;).

**Comment by saasasasasa** [comment]
> 
> 
> I'd be mildly intrigued about what a debug log might be saying about the cache size, for instance ;).

To be honest I've never used adb and logcat before, but I just run it with this command `adb logcat KOReader:I ActivityManager:* AndroidRuntime:* DEBUG:* *:F` which I found in some other thread and this is the log:
[logcat 2.txt](https://github.com/koreader/koreader/files/7123895/logcat.2.txt)

It took me like 30 minutes just to find out how to generate this :D. Is this alright, or a log with some other command is needed?

**Comment by pazos** [comment]
@saasasasasa: If you want/can install `pidcat` that's the most relevant one, as it includes inner logs and system logs for this app's pid.

If not then use https://github.com/koreader/koreader/blob/master/kodev#L1139

**But** you need to enable first debug logging in developer options (tools -> more tools -> dev opts)

**Comment by saasasasasa** [comment]
> **But** you need to enable first debug logging in developer options (tools -> more tools -> dev opts)

Oh, you mean in the app? I can't. When I install Koreader and launch it for the first time there is this loading screen with 3 dots and then it just goes immediately to the bug log screen.

**Comment by saasasasasa** [comment]
So I got it working by renaming the old "koreader" folder, uninstalling 04 version and installing the newest one. It created the new "koreader" folder and now it works fine. So I guess there is some files/settings in the old version that prevent the newer versions from working.

Although for now I'll stay on the old version until I figure out how to change the width of the bottom progress bar.

**Comment by saasasasasa** [comment]
So I got it working. The problem was with file `defaults.persistent.lua` I deleted it and Koreader v08 now works. And I got the progress bar width settings saved from the previous version so I'm happy.

This was the contents of the 'defaults.persistent.lua" file, so here was some problem:

-- For configuration changes that persists between updates
DGLOBAL_CACHE_SIZE_MINIMUM = 0
DKOPTREADER_CONFIG_CONTRAST = 0.3
DGLOBAL_CACHE_SIZE_MAXIMUM = 0
